src/js/Calculator.js

```diff
@@ -0,0 +1,103 @@
+import { BUTTON_TYPE, ERROR_MESSAGES, OPERATOR, VALUE } from "./constants.js";
+
+const $ = (target) => {
+  return document.querySelector(target);
+};
+
+class Calculator {
+  constructor() {
+    this.curNumber = "";
+    this.prevNumber = "";
+    this.operator = "";
+    this.init();
+  }
+
+  init() {
+    this.registerButtonClickHandler();
+  }
+
+  registerButtonClickHandler = () => {
+    const buttons = document.querySelectorAll("button");
+    const { DIGIT, OPERATION, MODIFIER } = BUTTON_TYPE;
+
+    buttons.forEach((button) => {
+      if (button.classList.contains(DIGIT)) {
+        button.addEventListener("click", () => this._onClickDigit(button));
+      }
+      if (button.classList.contains(OPERATION)) {
+        button.addEventListener("click", () => this._onClickOperator(button));
+      }
+      if (button.classList.contains(MODIFIER)) {
+        button.addEventListener("click", this._onClickModifier);
```

Comment:
```suggestion
      if (button.classList.contains(DIGIT)) {
        button.addEventListener("click", () => this._onClickDigit(button));
      }
      else if (button.classList.contains(OPERATION)) {
        button.addEventListener("click", () => this._onClickOperator(button));
      }
      else {
        button.addEventListener("click", this._onClickModifier);
```
`if .. else if .. else` 를 활용하는게 더 좋아보입니다. 왜냐하면 첫 번쨰에서 이벤트가 걸린 버튼은 굳이 두번쨰, 세번째 조건을 맞춰볼 필요가 없으니깐요 👍